cwms-data-api/src/main/java/cwms/cda/data/dto/timeseriesprofile/TimeSeriesProfileParser.java

```diff
@@ -0,0 +1,194 @@
+package cwms.cda.data.dto.timeseriesprofile;
+
+import java.math.BigInteger;
+import java.util.List;
+import java.util.Objects;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.databind.PropertyNamingStrategies;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.fasterxml.jackson.databind.annotation.JsonNaming;
+import com.fasterxml.jackson.databind.annotation.JsonPOJOBuilder;
+import cwms.cda.api.errors.FieldException;
+import cwms.cda.data.dto.CwmsDTO;
+import cwms.cda.formatters.Formats;
+import cwms.cda.formatters.annotations.FormattableWith;
+import cwms.cda.formatters.json.JsonV2;
+
+@FormattableWith(contentType = Formats.JSONV2, formatter = JsonV2.class)
+@JsonDeserialize(builder = TimeSeriesProfileParser.Builder.class)
+@JsonInclude(JsonInclude.Include.NON_NULL)
+@JsonNaming(PropertyNamingStrategies.KebabCaseStrategy.class)
+public class TimeSeriesProfileParser extends CwmsDTO
+{
+	private final String locationId;
+	private final String keyParameter;
+	private final char recordDelimiter;
+	private final char fieldDelimiter;
+	private final  String timeFormat;
+	private final String timeZone;
+	private final int timeField;
+	private final List<ParameterInfo> parameterInfo;
+	private final boolean timeInTwoFields;
+	protected TimeSeriesProfileParser(Builder builder)
+	{
+		super(builder.officeId);
+		locationId = builder.locationId;
+		keyParameter = builder.keyParameter;
+		recordDelimiter = builder.recordDelimiter;
+		fieldDelimiter = builder.fieldDelimiter;
+		timeFormat = builder.timeFormat;
+		timeZone = builder.timeZone;
+		timeField = builder.timeField;
+		parameterInfo = builder.parameterInfo;
+		timeInTwoFields = builder.timeInTwoFields;
+	}
+
+	@Override
+	public void validate() throws FieldException
+	{
+
+	}
+
+
+	public String getLocationId()
+	{
+		return locationId;
+	}
+
+	public String getKeyParameter()
+	{
+		return keyParameter;
+	}
+	public String getRecordDelimiter(){ return String.valueOf(recordDelimiter); }
+	public String getFieldDelimiter(){ return String.valueOf(fieldDelimiter); }
+	public List<ParameterInfo> getParameterInfo ()
+	{
+		return parameterInfo;
+	}
+
+	public String getTimeFormat()
+	{
+		return timeFormat;
+	}
+
+	public String getTimeZone()
+	{
+		return timeZone;
+	}
+
+	public BigInteger getTimeField()
+	{
+		return BigInteger.valueOf(timeField);
+	}
+
+	public String getTimeInTwoFields()
+	{
+		return timeInTwoFields?"T":"F";
```

Comment:
This doesnt follow established DTO Boolean handling present elsewhere in the CDA API

Reply:
changed to return boolean